### PR TITLE
Guess Who - Ekaterina 

### PR DESCRIPTION
**Pull request by klimenko3007**

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -248,96 +410,253 @@ const generateBoard = () => {
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
-
-// This function to start (and restart) the game
+const updateTracker = () => {
```
Comment (ettem) [inline code/script.js:413]: Perhaps comment the intention of this function.

File: code/script.js [hunk 2]
```diff
@@ -248,96 +410,253 @@ const generateBoard = () => {
...
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
+//Guess function allows the player to check the guess and secret person
+const guess = (suspect) => {
+  const userGuess = suspect;
+  playAlerSound()
```
Comment (ettem) [inline code/script.js:585]: playAlertSound instead?

File: code/script.js [hunk 3]
```diff
@@ -248,96 +410,253 @@ const generateBoard = () => {
...
+//Sound effects functions
+const playWinSound = () => {
+  const winSound = document.getElementById('win-sound');
+  winSound.play();
+}
+const playLooseSound = () => {
+  const looseSound = document.getElementById('loose-sound');
+  looseSound.play();
+}
+const playAlerSound = () => {
```
Comment (ettem) [inline code/script.js:626]: Concerning naming, same as line 585.

File: code/script.js [hunk 4]
```diff
@@ -248,96 +410,253 @@ const generateBoard = () => {
...
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+// Invokes the game timer function when website is loaded
+setInterval(gameTimer, 1000);
 
-  // Invoke a function to redraw the board with the remaining people.
-}
+// Event listeners
```
Comment (ettem) [inline code/script.js:634]: Good comments explaining your addEventListeners!

File: code/style.css [hunk 5]
```diff
@@ -43,20 +49,74 @@ select {
...
+.question-tracker {
+  color: white;
+  margin-top: 5px;
+  margin-bottom: 0;
+}
+.timer-wrapper {
+  margin-top: 5px;
+  color: white;
+
+}
```
Comment (ettem) [inline code/style.css:63]: move up to line 62

File: code/style.css [hunk 6]
```diff
@@ -43,20 +49,74 @@ select {
...
+.name-wrapper {
+  margin-top: 15px;
+  margin-bottom: 0;
+  color: white;
+}
+/* Board styling */
```
Comment (ettem) [inline code/style.css:69]: move down to line 70 adding a space between comments and previous grouped elements, like you've done before

File: code/style.css [hunk 7]
```diff
@@ -43,20 +49,74 @@ select {
 .guess-who-icon {
   width: 126px;
 }
+/* Aside section: player, timer, questions tracker  */
```
Comment (ettem) [inline code/style.css:52]: move down to line 53

File: code/style.css [hunk 8]
```diff
@@ -43,20 +49,74 @@ select {
...
+/* Board styling */
 
 .board-wrapper {
   width: 70%;
   display: flex;
   align-items: center;
   justify-content: center;
+  background-color: rgb(217, 217, 244);
+}
+/* Welcome conversation */
```
Comment (ettem) [inline code/style.css:78]: move down to add space on line 78

File: code/style.css [hunk 9]
```diff
@@ -43,20 +49,74 @@ select {
...
+  border-radius: 5px;
 }
+.welcome-button {
+  border-radius: 10px;
+  padding: 10px;
+  color: white;
+  border: var(--primary);
+  background-color: var(--primary);
 
+}
```
Comment (ettem) [inline code/style.css:112]: move up to line 111

File: code/style.css [hunk 10]
```diff
@@ -43,20 +49,74 @@ select {
 .guess-who-icon {
```
Comment (ettem) [inline code/style.css:49]: move up to line 48

File: code/style.css [hunk 11]
```diff
@@ -176,12 +237,29 @@ button {
 .win-or-lose h1 {
```
Comment (ettem) [inline code/style.css:237]: moving up to line 236

**Review by ettem — APPROVED** [review]
Katia, really nice work! Everything looks supernice and I can't find anything that doesn't work testing the game. I'm very impressed by the way you write your code and it really shows you have understood your javascript - also you finished all the requirements as far as I'm concerned, blue red and black, congratulations! As general comments I would like to say that your writing is really nice, indentations are good, it's logical, good structure and semantic tags are used. Your naming conventions are good as well and it is understandable easy-to-read logic in your functions. However you are on a different level than me so my suggestion is to add a few more comments in your code, especially the javascript, explaining for those who knows less than you :) Since I don't have any big remarks, rather a few suggestions only, I am approving. Keep up the ambitious work!

File: code/index.html [hunk 12]
```diff
@@ -1,64 +1,106 @@
...
+    <div class="win-or-lose">
+      <img class="guess-who-icon" src="images/guess-who-bubble.png" alt="Guess Who" />
+      <h1 id="winOrLoseText"></h1>
+      <button id="playAgain" class="filled-button win-loose">PLAY AGAIN</button>
+      <button id="new-player"class="filled-button win-loose">NEW PLAYER</button>
+    </div>
+  </section>
+  <audio id="win-sound" src="./images/win.mp3"></audio>
+  <audio id="loose-sound" src="./images/loose.mp3"></audio>
+  <audio id="alert-sound" src="./images/alert.mp3"></audio>
```
[the remark this answers is not in the document]
Comment (KarolCislowski) [reply code/index.html:102]: Win sound rocks 🥳

File: code/index.html [hunk 13]
```diff
@@ -1,64 +1,106 @@
...
+  <title>Guess Who by Ekaterina Klimenko</title>
+  <link rel="stylesheet" href="style.css" />
+  <link
+    href="https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap"
+    rel="stylesheet" />
+</head>
 
-      <h1>Does the person have</h1>
+<body>
+  <aside class="question-section rules" id="aside-rules">
```
[the remark this answers is not in the document]
Comment (KarolCislowski) [reply code/index.html:15]: Space separates 2 classes. You can assign as many classes to the element as you wish 🎨  

Inspect the Facebook page and you will see that every element will have many classes. 🥼 🔬 

There are bandles of classes ready to use as the famous Bootstrap.

File: code/script.js [hunk 14]
```diff
@@ -1,231 +1,393 @@
...
+const player = document.getElementById('player-name');
+const welcomeBoard = document.getElementById('welcome-section');
+const board = document.getElementById('board');
+const questions = document.getElementById('questions');
+const restartButton = document.getElementById('restart');
+const filterButton = document.getElementById('filter');
+const winOrLooseBoard = document.getElementById('winOrLose');
+const winOrLooseText = document.getElementById('winOrLoseText');
+const playAgainButton = document.getElementById('playAgain');
+const newPlayerButton = document.getElementById('new-player')
```
[the remark this answers is not in the document]
Comment (KarolCislowski) [reply code/script.js:16]: Please make your VScode add them for you. Go to the settings and search for semicolons. 🤓

**Review by KarolCislowski — APPROVED** [review]
This is a very impressive project! 🌈  You have shown great ambition 🦄

**Pull request closed without merging**